unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -341,10 +375,19 @@ public async UniTask<IReadOnlyList<string>> GetFriendsAsync(string userNameOrId,
         public void GetFriendRequests(int sentLimit, int sentSkip, int receivedLimit, int receivedSkip) =>
             apiBridge.GetFriendRequests(sentLimit, sentSkip, receivedLimit, receivedSkip);
 
-        public async UniTask<List<FriendRequest>> GetFriendRequestsAsync(int sentLimit, int sentSkip, int receivedLimit, int receivedSkip, CancellationToken cancellationToken)
+        public async UniTask<IEnumerable<FriendRequest>> GetFriendRequestsAsync(int sentLimit, int sentSkip, int receivedLimit, int receivedSkip, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
+            if (useSocialApiBridge)
+            {
+                var sent = outgoingFriendRequestsByTimestamp.Values.Skip(sentSkip).Take(sentLimit);
+                var received = incomingFriendRequestsByTimestamp.Values.Skip(receivedSkip).Take(receivedLimit);
+                var result = received.Concat(sent);
+
+                return result;
```

Comment:
It's error-prone to return a streaming iterator as every time it is accessed it will perform the logic starting from `Skip` and ending with `Concat` from the beginning - the possibility of multiple enumerations. And it would be best if you did not rely on that consumers do it only once.

Reply:
Fixed the code to match the proposal by @AjimenezDCL